gef-ui/server/controllers/amendments/what-needs-to-change/post-what-needs-to-change.test.ts

```diff
@@ -0,0 +1,248 @@
+import { STB_PIM_EMAIL } from '../../../constants/emails.ts';
+import { PortalFacilityAmendmentWithUkefIdMockBuilder } from '../../../../test-helpers/mock-amendment.ts';
+import { Deal } from '../../../types/deal.ts';
+import { getNextPage } from '../helpers/navigation.helper.ts';
+import { validateWhatNeedsToChange } from './validation.ts';
+import { ValidationError } from '../../../types/validation-error.ts';
+import { validationErrorHandler } from '../../../utils/helpers';
+import { PORTAL_AMENDMENT_PAGES } from '../../../constants/amendments.ts';
+
+jest.mock('../../../services/api', () => ({
+  getApplication: getApplicationMock,
+  getFacility: getFacilityMock,
+  updateAmendment: updateAmendmentMock,
+}));
+
+console.error = jest.fn();
+
+const dealId = 'dealId';
+const facilityId = 'facilityId';
+const amendmentId = 'amendmentId';
+
+const aMockError = () => new Error();
+
+const companyName = 'company name ltd';
+const userToken = 'userToken';
+
+const getHttpMocks = (amendmentOptions: string[] = ['changeCoverEndDate', 'changeFacilityValue']) =>
+  createMocks<PostWhatNeedsToChangeRequest>({
+    params: { dealId, facilityId, amendmentId },
+    session: {
+      user: { ...aPortalSessionUser(), roles: [ROLES.MAKER] },
+      userToken,
+      loginStatus: PORTAL_LOGIN_STATUS.VALID_2FA,
+    },
+    body: {
+      amendmentOptions,
+    },
+  });
+
+const mockDeal = { exporter: { companyName }, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED } as Deal;
+
+const mockFacility = {
+  hasBeenIssued: true,
+} as Facility;
+
+describe('postWhatNeedsToChange', () => {
+  let amendment: PortalFacilityAmendmentWithUkefId;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.spyOn(dtfsCommon, 'isPortalFacilityAmendmentsFeatureFlagEnabled').mockReturnValue(true);
+
+    amendment = new PortalFacilityAmendmentWithUkefIdMockBuilder().withDealId(dealId).withFacilityId(facilityId).withAmendmentId(amendmentId).build();
+
+    getApplicationMock.mockResolvedValue(mockDeal);
+    getFacilityMock.mockResolvedValue({ details: mockFacility });
+    updateAmendmentMock.mockResolvedValue(amendment);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken });
+  });
+
+  it('should call getFacility with the correct facilityId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    expect(getFacilityMock).toHaveBeenCalledTimes(1);
+    expect(getFacilityMock).toHaveBeenCalledWith({ facilityId, userToken: req.session.userToken });
```

Comment:
Please add

expect(console.error).not.toHaveBeenCalled();

Reply:
Have added 👍